### PR TITLE
Creating remaining DB tables and models

### PR DESCRIPTION
**Pull request by JesusGo115**
This is the Jira sprint for creating the remaining [tables](https://cascarita.atlassian.net/browse/C1-18?atlOrigin=eyJpIjoiNDQzM2FlZDVhNTg0NGQwMzllN2M1YTM3ZjY2NTY0OGEiLCJwIjoiaiJ9)

**Summary**:  Created 

- Divisions
- DivisionsHistories
- Fields
- FieldsHistories
- Games
- GamesHistories
- GamesStatuses
- Leagues
- LeaguesHistories
- Players
- PlayersHistories
- Seasons
- SeasonsHistories
- Sessions
- SessionsHistories
- Teams
- TeamsHistories
- TeamsPerformances
- TeamsPerformancesHistories
- TeamsSessions
- TeamsHistories

To get these tables in your DBeaver, cd to the server folder and run this command-line: `npx sequelize-cli db:migrate`

## Summary by CodeRabbit

- **New Features**
	- Added player creation functionality with validation for phone numbers.
	- Introduced player-related routes in the application.
	- Added new database tables for managing leagues, seasons, divisions, sessions, games, teams, and player associations, along with their historical data.
- **Bug Fixes**
	- Corrected naming inconsistencies by updating field names from camelCase to snake_case across various sections related to user registration and error handling.
- **Refactor**
	- Updated serialization of user data to improve consistency in data format.
	- Enhanced database schema with migrations for creating and updating tables to support new features and ensure data integrity.
- **Chores**
	- Added seeders for inserting initial soccer-related data into the database.

### CODE REVIEW

**Review by saulRM10 — CHANGES_REQUESTED** [review]

File: server/models/league.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+    static associate(models) {
+      // define association here
+      League.belongsTo(models.Group, { foreignKey: "groupId" });
+    }
+  }
+  League.init(
```
[the remark this answers is not in the document]
Comment (saulRM10) [reply server/models/league.js:15]: solved with migration

File: server/models/teamsdivisions.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      TeamsDivisions.belongsTo(models.Team, { foreignKey: "team_id" });
+      TeamsDivisions.belongsTo(models.Division, { foreignKey: "division_id" });
+    }
+  }
+  TeamsDivisions.init(
+    {
+      team_id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        references: {
```
[the remark this answers is not in the document]
Comment (saulRM10) [reply server/models/teamsdivisions.js]: No need for this table but we did take a look a associations in between out tables and updates models accordingly

File: server/models/session.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      allowNull: false
+    },
+    league_id: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+    },
+    },
+    {
+    sequelize,
+    modelName: 'Session',
```
[the remark this answers is not in the document]
Comment (saulRM10) [reply server/models/session.js]: all done

File: server/models/teamssession.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  class TeamsSession extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+    static associate(models) {
+      // define association here
+      TeamsSession.belongsTo(models.Session, { foreignKey: "session_id" });
+      TeamsSession.belongsTo(models.Team, { foreignKey: "team_id" });
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/models/teamssession.js]: Solved by changing the relationships of a handful of tables that required it (i.e. TeamsSessions, Games, PlayersTeams, etc.)

File: server/migrations/20240223200359-update-group-table-column-names-snake-case.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.renameColumn("Groups", "updatedAt", "updated_at");
+    await queryInterface.renameColumn("Groups", "createdAt", "created_at");
+
+    await queryInterface.changeColumn("Groups", "name", {
+      type: Sequelize.STRING,
+      allowNull: false,
+      unique: true,
+    });
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/migrations/20240223200359-update-group-table-column-names-snake-case.js:13]: There's no reason to roll back this migration

File: server/seeders/20240227210651-insert-soccer-data.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  },
+
+  async down (queryInterface, Sequelize) {
+    /**
+     * Add commands to revert seed here.
+     *
+     * Example:
+     * await queryInterface.bulkDelete('People', null, {});
+     */
+  }
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/seeders/20240227210651-insert-soccer-data.js:24]: Building blocks for the feature we will be tackling.

File: server/migrations/20240226214443-drop-team-divisions.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.dropTable("TeamsDivisions");
+  },
+
+  async down(queryInterface, Sequelize) {},
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/migrations/20240226214443-drop-team-divisions.js:9]: This table is not needed.

File: server/migrations/20240226214706-drop-team-divisions-history.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.dropTable("TeamsDivisionsHistories");
+  },
+
+  async down(queryInterface, Sequelize) {},
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/migrations/20240226214706-drop-team-divisions-history.js:9]: This table is no longer needed.

File: server/migrations/20240222212730-drop-plyaers-table.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.dropTable("Plyaers");
+  },
+
+  async down(queryInterface, Sequelize) {},
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/migrations/20240222212730-drop-plyaers-table.js:9]: The Plyaers table is no needed.

File: server/app.js [hunk 10]
```diff
@@ -10,6 +10,7 @@ const bodyParser = require("body-parser");
 const GroupRoutes = require("./routes/group.routes");
 const RoleRoutes = require("./routes/role.routes");
 const UserRoutes = require("./routes/user.routes");
+const PlayerRoutes = require("./routes/player.routes");
 
 const app = express();
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/app.js:16]: Updated the app.js with these new lines of code.

File: server/modelsHistory/gameshistory.js [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    },
+    {
+      sequelize,
+      modelName: "GamesHistory",
+      createdAt: "created_at",
+      updatedAt: "updated_at",
+    }
+  );
+  return GamesHistory;
+};
```
[the remark this answers is not in the document]
Comment (JesusGo115) [reply server/modelsHistory/gameshistory.js]: Updated the GamesHistories table with the suggested fixes.

**Review by saulRM10 — APPROVED** [review]
Looks good, @JesusGo115  and @Jen-464  great job with building the tables and model and checking the associations between them

**Pull request merged**